src/parser/WASMParser.cpp

```diff
@@ -1288,7 +1288,7 @@ class WASMBinaryReader : public wabt::WASMBinaryReaderDelegate {
             // if there is local.set code ahead,
             // we can use local variable position as expr target position
             auto localSetInfo = readAheadLocalGetIfExists();
-            if (localSetInfo.first) {
+            if (localSetInfo.first && !m_localInfo.empty()) {
```

Comment:
Is this a bugfix in the parser?

Reply:
Yes. I will investigate it further and make a patch for it because it is not directly connected to this patch.